api/app/controllers/transaction.controller.ts

```diff
@@ -279,3 +281,48 @@ export const getAllTransactions = async (req: Request, res: Response) => {
     res.status(500).json({ message: "Internal server error" });
   }
 };
+
+export const processUnpaidReviewTransaction = async (
+  req: Request,
+  res: Response
+) => {
+  const reviewId = req.body.reviewId;
+  if (!reviewId) {
+    return res.status(400).send({ message: "Review id is required" });
+  }
+  try {
+    const review = await Review.findByPk(reviewId);
+    if (!review) {
+      return res
+        .status(404)
+        .send({ message: `Review with id=${reviewId} not found` });
+    }
+    const unpaidReviewTransaction = await Transaction.findOne({
+      where: {
+        reviewId: reviewId,
+      },
+    });
+    if (unpaidReviewTransaction) {
+      return res
+        .status(200)
+        .send({ message: `Transaction for review - ${reviewId} already exists` });
+    }
+    const associatedTranscript = await Transcript.findByPk(review.transcriptId);
+    if (!associatedTranscript) {
+      return res.status(404).send({
+        message: `Transcript with id=${review.transcriptId} not found`,
+      });
+    }
+    await addCreditTransactionQueue(associatedTranscript, review);
```

Comment:
I am thinking we can use sequelize transactions here

Reply:
Why? This function is executed in a worker as a cron job. I don't see why that's necessary and how thats possible.

Reply:
Yea, missed where the function is running. We can go ahead